File: week1/problemset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
int calcualte_coins_amount(int coin_value, int amount);

int main(void)
{
    int owed;

    do
    {
        owed = get_int("Change owed: ");
    }
    while(owed < 1);

    int total_coins = 0;

    if (owed >= 25)
    {
        int quarters = calcualte_coins_amount(25,owed);
        owed = owed % (quarters * 25);
        total_coins += quarters;
    }

    if (owed >= 10)
    {
        int dimes = calcualte_coins_amount(10, owed);
        owed = owed % (dimes * 10);
        total_coins += dimes;
    }

    if (owed >= 5)
    {
        int nickels = calcualte_coins_amount(5, owed);
        owed = owed % (nickels * 5);
        total_coins += nickels;
    }

    if (owed >= 1)
    {
        int pennies = calcualte_coins_amount(1, owed);
        owed = owed % (pennies * 1);
        total_coins += pennies;
    }

    printf("%i", total_coins);
}

int calcualte_coins_amount(int coin_value, int amount)
{
    return amount / coin_value;
}

File: week6/problemset/sentimental-cash/cash.py
def main():
    change = get_change()

    total_coins = 0

    if change >= 25:
        quarters = calculate_coins_amount(25, change)
        change %= (quarters * 25)
        total_coins += quarters

    if change >= 10:
        dimes = calculate_coins_amount(10, change)
        change %= (dimes * 10)
        total_coins += dimes

    if change >= 5:
        nickels = calculate_coins_amount(5, change)
        change %= (nickels * 5)
        total_coins += nickels

    if change >= 1:
        pennies = calculate_coins_amount(1, change)
        change %= pennies
        total_coins += pennies

    print(total_coins)


def calculate_coins_amount(coin_value, amount):
    return int(amount / coin_value)


def get_change():
    while True:
        change = get_float("Change: ")

        if change > 0:
            break

    return change * 100


def get_float(prompt):
    while True:
        try:
            return float(input(prompt))
        except ValueError:
            print("Not a Number")


main()
